From cc14e8be8f7734675e57796b6a2b45e90a16d621 Mon Sep 17 00:00:00 2001
From: Billy <xiaobing.feng@windriver.com>
Date: Wed, 23 Jan 2013 19:15:06 +0800
Subject: Storage: Correct Low volume alert message

DeviceStorageMonitorService monitors internal storage. When the
free memory in the internal storage gets below the low threshold,
it sends a message to "Settings" app to update storage devices
information with alarm message. Currently, the code not only
updates the internal storage information, but also all external
storage units ( SD and USB), even they still have much free memory
or even are not inserted. This patch makes the internal storage
related alarm appear only in the internal storage section.

Issue: AXIA-1058
Category: aosp improvement
Domain: AOSP-Storage
Origin: internal
Upstream-Candidate: yes

Change-Id: I9c197e8922e13e30a3456a9eff417e93e536ae11
Signed-off-by: Xiaobing Feng <xiaobing.feng@windriver.com>
---
 .../StorageVolumePreferenceCategory.java           |   25 +++++++++++--------
 1 files changed, 14 insertions(+), 11 deletions(-)

diff --git a/src/com/android/settings/deviceinfo/StorageVolumePreferenceCategory.java b/src/com/android/settings/deviceinfo/StorageVolumePreferenceCategory.java
index ed5085a..1a1db51 100644
--- a/src/com/android/settings/deviceinfo/StorageVolumePreferenceCategory.java
+++ b/src/com/android/settings/deviceinfo/StorageVolumePreferenceCategory.java
@@ -214,18 +214,21 @@ public class StorageVolumePreferenceCategory extends PreferenceCategory {
         }
 
         final IPackageManager pm = ActivityThread.getPackageManager();
-        try {
-            if (pm.isStorageLow()) {
-                mStorageLow = new Preference(context);
-                mStorageLow.setOrder(ORDER_STORAGE_LOW);
-                mStorageLow.setTitle(R.string.storage_low_title);
-                mStorageLow.setSummary(R.string.storage_low_summary);
-                addPreference(mStorageLow);
-            } else if (mStorageLow != null) {
-                removePreference(mStorageLow);
-                mStorageLow = null;
+        // Only show low volume messge for internal storage
+        if (mVolume == null) {
+            try {
+                if (pm.isStorageLow()) {
+                    mStorageLow = new Preference(context);
+                    mStorageLow.setOrder(ORDER_STORAGE_LOW);
+                    mStorageLow.setTitle(R.string.storage_low_title);
+                    mStorageLow.setSummary(R.string.storage_low_summary);
+                    addPreference(mStorageLow);
+                } else if (mStorageLow != null) {
+                    removePreference(mStorageLow);
+                    mStorageLow = null;
+                }
+            } catch (RemoteException e) {
             }
-        } catch (RemoteException e) {
         }
     }
 
-- 
1.7.4.1

